crates/blockchain/payload.rs

```diff
@@ -218,6 +218,7 @@ impl PayloadBuildContext {
         payload: Block,
         storage: &Store,
         blockchain_type: BlockchainType,
+        fee_vault: Option<Address>,
```

Comment:
We should probably split this into `new_for_l1` and `new_for_l2`, just to keep interfaces clean:

```rust
fn new(...) { /* same as now */ }
pub fn new_for_l1(...) { new(...) }
pub fn new_for_l2(...) { new(...) }
```

We could also remove both `blockchain_type` and `fee_vault` entirely, creating the VM in the wrappers and passing it on to `new` directly.